bindings/wasm/manifold.d.ts

```diff
@@ -12,474 +12,32 @@
- * calculated by the static Defaults.
- */
-function revolve(crossSection: Polygons, circularSegments?: number): Manifold;
-
-function union(a: Manifold, b: Manifold): Manifold;
-function difference(a: Manifold, b: Manifold): Manifold;
-function intersection(a: Manifold, b: Manifold): Manifold;
-
-function union(manifolds: Manifold[]): Manifold;
-function difference(manifolds: Manifold[]): Manifold;
-function intersection(manifolds: Manifold[]): Manifold;
-
-/**
- * Constructs a new manifold from a list of other manifolds. This is a purely
- * topological operation, so care should be taken to avoid creating
- * overlapping results. It is the inverse operation of Decompose().
- *
- * @param manifolds A list of Manifolds to lazy-union together.
- */
-function compose(manifolds: Manifold[]): Manifold;
-
-/**
- * Constructs a level-set Mesh from the input Signed-Distance Function (SDF).
- * This uses a form of Marching Tetrahedra (akin to Marching Cubes, but better
- * for manifoldness). Instead of using a cubic grid, it uses a body-centered
- * cubic grid (two shifted cubic grids). This means if your function's interior
- * exceeds the given bounds, you will see a kind of egg-crate shape closing off
- * the manifold, which is due to the underlying grid.
- *
- * @param sdf The signed-distance function which returns the signed distance of
- * a given point in R^3. Positive values are inside, negative outside.
- * @param bounds An axis-aligned box that defines the extent of the grid.
- * @param edgeLength Approximate maximum edge length of the triangles in the
- * final result. This affects grid spacing, and hence has a strong effect on
- * performance.
- * @param level You can inset your Mesh by using a positive value, or outset
- * it with a negative value.
- */
-function levelSet(
-    sdf: (point: Vec3) => number, bounds: Box, edgeLength: number,
-    level?: number): Manifold;
-
-/**
- * @name Defaults
- * These static properties control how circular shapes are quantized by
- * default on construction. If circularSegments is specified, it takes
- * precedence. If it is zero, then instead the minimum is used of the segments
- * calculated based on edge length and angle, rounded up to the nearest
- * multiple of four. To get numbers not divisible by four, circularSegments
- * must be specified.
- */
-///@{
-function setMinCircularAngle(angle: number): void;
-function setMinCircularEdgeLength(length: number): void;
-function setCircularSegments(segments: number): void;
-function getCircularSegments(radius: number): number;
-///@}
+import * as T from './manifold-encapsulated-types';
+export * from './manifold-global-types';
 
-/**
- * Returns the first of n sequential new unique mesh IDs for marking sets of
- * triangles that can be looked up after further operations. Assign to
- * Mesh.runOriginalID vector.
- */
-function reserveIDs(count: number): number;
+export type Manifold = T.Manifold;
+export type Mesh = T.Mesh;
```

Comment:
So, do we actually want these? I thought this was what you removed so that `new Manifold()` would correctly error?

Reply:
The way it was done before, both the type and the value are exported. In the new way, Manifold and Mesh are re-exported only as a type (which is why it has `export type Manifold` instead of `export class Manifold`), so if you try to do `new Manifold` you will now get an error. It's only a type AND value when used from a ManifoldStatic instance (which is why it's using `Manifold: typeof T.Manifold` in ManifoldStatic)

Reply:
Ah, thank you for the explanation!